presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -547,14 +547,12 @@ private Visitor(Session session)
         @Override
         public PhysicalOperation visitRemoteSource(RemoteSourceNode node, LocalExecutionPlanContext context)
         {
-            // Having multiple remote ExchangeClients per outputId is not acceptable for an OutputBuffer (e.g. SharedBuffer).
-            // Setting the driver instance count to 1 here:
-            // * validates that it was either unset or set to 1
-            // * pins it so that it can not later be set to a different value
-            context.setDriverInstanceCount(1);
-
             List<Type> types = getSourceOperatorTypes(node, context.getTypes());
 
+            if (!context.getDriverInstanceCount().isPresent()) {
+                context.setDriverInstanceCount(getTaskConcurrency(session));
```

Comment:
Sorry, I still don't understand how this code works but all I wanted to know is, whether this change increases number of hash partitioners in simple queries like
```
set session task_concurrency=16;
select avg(totalprice) from orders group by clerk;
```
for which hash partitioning in task doing aggregation was a bottleneck.